sygnal/helper/context_factory.py

```diff
@@ -125,22 +129,19 @@ class ConnectionVerifier:
 
     # This code is based on twisted.internet.ssl.ClientTLSOptions.
 
-    def __init__(self, hostname):
-        if isIPAddress(hostname) or isIPv6Address(hostname):
-            self._hostnameBytes = hostname.encode("ascii")
+    def __init__(self, hostname: bytes):
+        _decoded = hostname.decode("ascii")
+        if isIPAddress(_decoded) or isIPv6Address(_decoded):
```

Comment:
Confusingly these functions [will accept bytes](https://github.com/twisted/twisted/blob/525da4a0becad1e814c74f12c365f15fe9cd10de/src/twisted/internet/abstract.py#L487-L503), even though the annotations disallow that.